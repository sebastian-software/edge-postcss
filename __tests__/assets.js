import { compile, compileSameFolder } from "./core"

// ====================================================
// ================= URLS/ASSETS ======================
// ====================================================

test("Asset URL", () =>
  compile(".icon { background: url('./fixtures/formula.png'); }")
)

test("Asset URL (Same Folder)", () =>
  compileSameFolder(".icon { background: url('./formula.png'); }")
)

test("Asset Size", () =>
  compile(".icon { background-size: width('./fixtures/formula.png') height('./fixtures/formula.png'); }")
)

test("Asset Size (Same Folder)", () =>
  compileSameFolder(".icon { background-size: width('./formula.png') height('./formula.png'); }")
)

test("Import with Asset URL", () =>
  compile("@import './fixtures/import-c.css';")
)

test("Import with SVG URL", () =>
  compile("@import './fixtures/other/Home.css';")
)

test("Import with Webfonts", () =>
  compile("@import './fixtures/font/SourceSansPro.css';")
)
