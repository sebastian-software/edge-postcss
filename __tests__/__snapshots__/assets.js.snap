// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asset Size (Same Folder) 1`] = `
".icon{
background-size:304px 85px
}"
`;

exports[`Asset Size 1`] = `
".icon{
background-size:304px 85px
}"
`;

exports[`Asset URL (Same Folder) 1`] = `
".icon{
background:url(./formula.png)
}"
`;

exports[`Asset URL 1`] = `
".icon{
background:url(../fixtures/formula.png)
}"
`;

exports[`Import with Asset URL 1`] = `
".image{
background:url(../fixtures/formula.png)
}"
`;

exports[`Import with SVG URL 1`] = `
".preloader{
background:url(../fixtures/other/logo.svg);
border:1px solid #000
}"
`;

exports[`Import with Webfonts 1`] = `
"@font-face{
src:url(../fixtures/font/SourceSansPro-Light.otf.woff2) format(\\"woff2\\"),url(../fixtures/font/SourceSansPro-Light.otf.woff) format(\\"woff\\");
font-family:Source Sans Pro;
font-weight:300;
font-style:normal
}"
`;
